mobile/src/main/java/org/openhab/habdroid/ui/ItemPickerActivity.java

```diff
@@ -0,0 +1,275 @@
+        searchView.setOnQueryTextListener(this);
+
+        return true;
+    }
+
+    public boolean onOptionsItemSelected(MenuItem item) {
+        Log.d(TAG, "onOptionsItemSelected()");
+        if (item.getItemId() == android.R.id.home) {
+            finish(false, null, null);
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    private void loadItems() {
+        Connection connection;
+        try {
+            connection = ConnectionFactory.getUsableConnection();
+        } catch (ConnectionException e) {
+            connection = null;
+        }
+        if (connection == null) {
+            updateViewVisibility(false, true);
+            return;
+        }
+
+        mItemPickerAdapter.clear();
+        updateViewVisibility(true, false);
+
+        final AsyncHttpClient client = connection.getAsyncHttpClient();
+        mRequestHandle = client.get("rest/items", new AsyncHttpClient.StringResponseHandler() {
+            @Override
+            public void onSuccess(String responseBody, Headers headers) {
+                try {
+                    ArrayList<Item> items = new ArrayList<>();
+                    JSONArray jsonArray = new JSONArray(responseBody);
+                    for (int i = 0; i < jsonArray.length(); i++) {
+                        JSONObject itemJson = jsonArray.getJSONObject(i);
+                        Item item = Item.fromJson(itemJson);
+                        if (!item.readOnly()) {
+                            items.add(item);
+                        }
+                    }
+                    Log.d(TAG, "Item request success, got " + items.size() + " items");
+                    mItemPickerAdapter.setItems(items);
+                    handleInitialHighlight();
+                    updateViewVisibility(false, false);
+                } catch (JSONException e) {
+                    Log.d(TAG, "Item response could not be parsed", e);
+                    updateViewVisibility(false, true);
+                }
+            }
+
+            @Override
+            public void onFailure(Request request, int statusCode, Throwable error) {
+                updateViewVisibility(false, true);
+                Log.e(TAG, "Item request failure", error);
+            }
+        });
+    }
+
+    @Override
+    public void onClick(View view) {
+        if (view == mRetryButton) {
+            loadItems();
+        }
+    }
+
+    @Override
+    public void onItemClicked(Item item) {
+        if (item == null) {
+            return;
+        }
+
+        SuggestCommandsFactory suggestCommandsFactory =
+                new SuggestCommandsFactory(this, true);
+        suggestCommandsFactory.fill(item);
+
+        List<String> labels = suggestCommandsFactory.getLabels();
+        List<String> commands = suggestCommandsFactory.getCommands();
+
+        labels.add(getString(R.string.item_picker_custom));
```

Comment:
This shouldn't be shown if makes no sense, because `commands` already contains all possible states, e.g. for Switches.